app/models/dataset/DataStore.scala

```diff
@@ -114,7 +114,7 @@ class DataStoreDAO @Inject()(sqlClient: SqlClient)(implicit ec: ExecutionContext
     for {
       accessQuery <- readAccessQuery
       r <- run(q"SELECT $columns FROM $existingCollectionName WHERE name = $name AND $accessQuery".as[DatastoresRow])
-      parsed <- parseFirst(r, name)
+      parsed <- parseFirst(r, name) ?~> "dataStore.notFound"
```

Comment:
I wouldn’t add notFound to parseFirst. I think parseFirst has its own error message, like `s"Parsing failed for row in $collectionName queried by $queryLabel"`
```suggestion
      parsed <- parseFirst(r, name)
```